packages/dds/tree/src/schema-stored/storedSchemaRepository.ts

```diff
@@ -71,22 +56,16 @@ export class StoredSchemaRepository<TPolicy extends SchemaPolicy = SchemaPolicy>
      * Combined with support for such namespaces in the allowed sets in the schema objects,
      * that might provide a decent alternative to extraFields (which is a bit odd).
      */
-    public constructor(
-        public readonly policy: TPolicy,
-        data?: SchemaData,
-    ) {
+    public constructor(public readonly policy: TPolicy, data?: SchemaData) {
         super();
-        if (data !== undefined) {
-            this.data.treeSchema = new Map(data.treeSchema);
-            this.data.globalFieldSchema = new Map(data.globalFieldSchema);
-        }
+        this.data = {
+            treeSchema: new Map(data?.treeSchema ?? []),
```

Comment:
If data isn't provided, is the `[]` necessary?

Reply:
no actually. Passing undefined to the map constructor is accepted and produces and empty map. Optional arguments are strange in typescript...

Fixed.